### ISSUE TITLE
Contribution Policy

### ISSUE THREAD

**Issue by techfort** [techfort/skrymir#7]
Proposed language spec:
- asm.js for low-level code (full-array iterations, integer / float operations etc.)
- ES6 features whitelist:
  1. let
  2. const
  3. generators
  4. template strings
  5. destructuring
  6. tail calls
  7. map/set/weakmap/weakset

Frameworks:
- browserify
- karma/jasmine
- istanbul 

Code style:
https://github.com/airbnb/javascript

Feel free to add.

**Comment by techfort** [comment]
Regarding Babel, how optimized is the generated code?

**Comment by seriousme** [comment]
The question is not: how optimized is the code, but how optimizable :-)
My answer: no idea ;-)
However we can expect that pretty soon all engines will have native support for the mainstream ES6 stuff. So e.g. if people are trying to wrestle the last cycle out of their Loki install they better use iojs 1.7 than node 0.6..

**Comment by techfort** [comment]
Sure, I suppose it'll be while before Loki 2.0 is out anyway. A lot may
change.

**Comment by seriousme** [comment]
Darn, wish we could use [ES6 Comprehensions](https://babeljs.io/docs/learn-es6/#comprehensions) as that would enable 

```
var results = [
  for(c of customers)
    if (c.city == "Seattle")
      { name: c.name, age: c.age }
]
```

which would be a quite powerful way to define a query :-)
(to see it in action check out [this](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&playground=false&code=var%20customers%20%3D%20%5B%7Bcountry%3A%20'Ireland'%2C%20name%3A'Joe'%2C%20company%3A'Techfort'%7D%5D%0D%0Avar%20results%20%3D%20%5B%0D%0A%20%20for%28c%20of%20customers%29%0D%0A%20%20%20%20if%20%28c.company%20%3D%3D%20%22Techfort%22%29%0D%0A%20%20%20%20%20%20%7B%20name%3A%20c.name%2C%20state%3A%20c.country%20%7D%0D%0A%5D%0D%0Aconsole.log%28results%29%3B%0D%0A))

Hard part of course being able to apply index lookups to preselect the candidates for the result set. 
(one would need to parse the AST of the query to figure out what props are being matched)
Would be really cool (but out of my league ;-)) to make this work :-)

**Comment by techfort** [comment]
Yeah list comprehensions are one of the things i love about python. Having said that, it looks to me that for the moment it's more syntactic sugar than performance improvements. Unless this is something that at some point involves TCO in which case it will perform much better when ES6 is stable.
